src/model/document.ts

```diff
@@ -12,7 +12,7 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
-import {Analyzer} from '../analyzer';
+import {AnalyzerCacheContext} from '../analyzer';
```

Comment:
It was in `../analyzer` alongside `Analyzer` but that's not super clean. I broke it out into its own file and created a new directory `core` for code that implements the core abstract pieces of the analyzer.